router/handle_lifecycle.go

```diff
@@ -118,8 +120,12 @@ func (rt *Handle) Setup(
 	rt.routerResponseTransformStat = stats.Default.NewTaggedStat("response_transform_latency", stats.TimerType, statTags)
 	rt.throttlingErrorStat = stats.Default.NewTaggedStat("router_throttling_error", stats.CountType, statTags)
 	rt.throttledStat = stats.Default.NewTaggedStat("router_throttled", stats.CountType, statTags)
-
-	rt.transformer = transformer.NewTransformer(rt.netClientTimeout, rt.transformerTimeout)
+	cache := v2.NewCache()
+	oauthLock := rudderSync.NewPartitionRWLocker()
+	rt.transformer = transformer.NewTransformer(rt.netClientTimeout, rt.transformerTimeout, cache, oauthLock,
+		backendConfig, &rt.reloadableConfig.oauthV2Enabled,
+		transformer.WithExpirationTimeDiff(rt.reloadableConfig.oauthV2ExpirationTimeDiff),
```

Comment:
We probably don't need function options here. We can simply create oauthV2ExpirationTimeDiff within the constructor (`NewTransformer`) itself.

```GO
func NewTransformer(destinationTimeout, transformTimeout time.Duration, backendConfig backendconfig.BackendConfig, oauthV2Enabled *misc.ValueLoader[bool]) Transformer {
	handle := &handle{
		oauthV2EnabledLoader: oauthV2Enabled,
		expirationTimeDiff:   config.GetReloadableDurationVar(5, time.Minute, "Router."+rt.destType+".oauth.expirationTimeDiff", "Router.oauth.expirationTimeDiff"),
	}
	handle.setup(destinationTimeout, transformTimeout, backendConfig)
	return handle
}
```

Reply:
Yes. Done